cluster-autoscaler/simulator/dynamicresources/utils/resource_claims_test.go

```diff
@@ -0,0 +1,735 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	apiv1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestClaimOwningPod(t *testing.T) {
+	truePtr := true
+	for _, tc := range []struct {
+		testName string
```

Comment:
Personally I prefer `map[]struct` because it ensures at compile time that I do not reuse the same test name twice.

Also, better avoid spaces in test names. They get replaced with underscores by `go test` and then the tests listed in the output can't be found easily in the source code. `go test -run=...` also is more complicated to get right.

Reply:
> benefits of descriptive test names outweigh the downsides

They can still be as descriptive as they are now, just don't use spaces as separators. If it's not readable without them, then it's also not readable in the `go test` output.

> I'd leave it for consistency

Consistency certainly trumps my personal preferences, :+1: